### PR TITLE
Crafting display infinite

### LINKED ISSUE #33857: Crafting with infinite water shows int max value
# Describe the bug

When crafting with an infinite water source (a well in my case) the amount of water shown is 2^31-1. I think this looks bad and should say something like "infinite" or "from well."

# Expected behavior

Display water (1 of infinite) or water (1 from well) or water (1) or just water, others will have a better idea of what this should say to look good, 2147483647 just is not it.

# Screenshots

Screenshot 2019-09-06 at 21 05 21

# Versions and configuration

 - OS: ChromeOS 76
 - Game Version: 0.D-7343-g8333e52d
 - Graphics version: Tiles
 - Mods loaded: None

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
```SUMMARY: Bugfixes "Crafting with infinite water shows int max value"```

#### Purpose of change
Fixes #33857

#### Describe the solution
Add check in to_string method to show the word infinite instead of int32max

### CODE REVIEW

File: src/requirements.cpp [hunk 1]
```diff
@@ -125,7 +125,11 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == INT32_MAX ) {
+            return string_format( npgettext( "requirement", "%2$d %1$s of infinite", "%2$d %1$s of infinite",
```
Comment (Qrox) [inline src/requirements.cpp]:
```suggestion
            //~ %1$s: item name, %2$d: charge requirement
            return string_format( npgettext( "requirement", "%1$s (%2$d of infinite)", "%1$s (%2$d of infinite)",
```
To be consistent with the code below.

File: src/requirements.cpp [hunk 2]
```diff
@@ -125,7 +125,11 @@ std::string item_comp::to_string( const int batch, const int avail ) const
...
+                                             c ),
+                                  type_ptr->nname( c ), c );
```
Comment (Qrox) [inline src/requirements.cpp]:
```suggestion
                                  type_ptr->nname( 1 ), c );
```
Singular form is always used for items counted by charges

File: src/requirements.cpp [hunk 3]
```diff
@@ -125,7 +125,12 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == INT32_MAX ) {
```
Comment (ZhilkinSerg) [inline src/requirements.cpp]: What if `INT32_MAX` is not available during compile time?

File: src/requirements.cpp [hunk 4]
```diff
@@ -125,7 +125,13 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == 2147483647/* int32 max */ ) {
```
Comment (anothersimulacrum) [inline src/requirements.cpp]: If you're going to use this twice, assign it to a variable, so it's not just a magic number.

Comment (anothersimulacrum) [reply src/requirements.cpp]:
```suggestion
        if( avail == std::numeric_limits<int>::max() ) {
```
Alternatively 
(The appropriate header is included)

File: src/requirements.cpp [hunk 5]
```diff
@@ -125,7 +125,13 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == std::numeric_limits<int>::max() ) {
```
Comment (Qrox) [inline src/requirements.cpp]: It's probably better to use `item::INFINITE_CHARGES` here for consistency with the rest of the crafting code.

File: src/requirements.cpp [hunk 6]
```diff
@@ -135,7 +141,12 @@ std::string item_comp::to_string( const int batch, const int avail ) const
                                   type_ptr->nname( 1 ), c );
         }
     } else {
-        if( avail > 0 ) {
+        //~ %1$s: item name, %2$d: charge requirement
```
Comment (Qrox) [inline src/requirements.cpp]:
```suggestion
```

File: src/requirements.cpp [hunk 7]
```diff
@@ -135,7 +141,12 @@ std::string item_comp::to_string( const int batch, const int avail ) const
...
+        if( avail == std::numeric_limits<int>::max() ) {
+            return string_format( npgettext( "requirement", "%2$d %1$s of infinite", "%2$d %1$s of infinite",
```
Comment (Qrox) [inline src/requirements.cpp]:
```suggestion
            //~ %1$s: item name, %2$d: required count
            return string_format( npgettext( "requirement", "%2$d %1$s of infinite", "%2$d %1$s of infinite",
```

File: src/requirements.cpp [hunk 8]
```diff
@@ -135,7 +141,12 @@ std::string item_comp::to_string( const int batch, const int avail ) const
                                   type_ptr->nname( 1 ), c );
         }
     } else {
-        if( avail > 0 ) {
+        if( avail == std::numeric_limits<int>::max() ) {
```
Comment (Qrox) [inline src/requirements.cpp]:
```suggestion
        if( avail == item::INFINITE_CHARGES ) {
```

**Pull request merged**